### PR TITLE
feat(TU-9582): Export correct prop types for react components

### PR DESCRIPTION
**Pull request by leandro-serafim**
Exported correct types for reacting components props.

I used this wrapper as example to test it.

```
import React from 'react'
import { Popover, PopoverProps, PopupButton, PopupButtonProps, Sidetab, SidetabProps, SliderButton, SliderButtonProps, Widget, WidgetProps } from '@typeform/embed-react'

type EmbedReactProps =
  | { type: 'popover' } & PopoverProps
  | { type: 'popup' } & PopupButtonProps
  | { type: 'sidetab' } & SidetabProps
  | { type: 'slider' } & SliderButtonProps
  | { type: 'widget' } & WidgetProps

const EmbedReact = ({ type, ...rest }: EmbedReactProps) => {
  switch (type) {
    case 'popover':
      return <Popover {...rest} />
    case 'popup':
      return <PopupButton {...rest} />
    case 'sidetab':
      return <Sidetab {...rest} />
    case 'slider':
      return <SliderButton {...rest} />
    case 'widget':
      return <Widget {...rest} />
    default:
      return null
  }
}

export default function renderAll() {
  return <>
    <EmbedReact type="popover" id="my-popover" />
    <EmbedReact type="popup" id="my-popup-button">test</EmbedReact>
    <EmbedReact type="sidetab" id="my-sidetab" />
    <EmbedReact type="slider" id="my-slider-button" />
    <EmbedReact type="widget" id="my-popup-button" />
  </>
}

```

Fixes: #643

### CODE REVIEW

**Comment by typeform-ops-gha** [comment]
*[BOT]* Preview available with hash `63e66e24cfb40d28e723215c5eb58cc23835bcee` [here](https://embed.typeform.com/63e66e24cfb40d28e723215c5eb58cc23835bcee/embed.js).

**Comment by typeform-ops-gha** [comment]
*[BOT]* Preview available with hash `b2d57551435cd255a69c5ec51415cbc393eee2ba` [here](https://embed.typeform.com/b2d57551435cd255a69c5ec51415cbc393eee2ba/embed.js).

**Comment by typeform-ops-gha** [comment]
*[BOT]* Preview available with hash `199d66875ddb83ffc7b77c5f7fb16645922ff9c0` [here](https://embed.typeform.com/199d66875ddb83ffc7b77c5f7fb16645922ff9c0/embed.js).

File: packages/embed-react/src/index.tsx [hunk 1]
```diff
@@ -9,17 +9,27 @@ import {
...
 
-const Widget = memoComponent(WidgetComponent)
-
-const PopupButton = memoComponent(makeButtonComponent<PopupOptions>(createPopup, 'popup'))
+export { WidgetProps }
+export const Widget = memoComponent<WidgetProps>(WidgetComponent)
 
-const SliderButton = memoComponent(makeButtonComponent<SliderOptions>(createSlider, 'slider'))
+export type PopupButtonProps = ButtonComponentProps<PopupOptions>
+export const PopupButton = memoComponent<PopupButtonProps>(makeButtonComponent<PopupOptions>(createPopup, 'popup'))
```
Comment (leandro-serafim) [inline packages/embed-react/src/index.tsx:26]: Export type and component.

**Comment by mathio** [comment]
Unit tests fail as they do not allow `aria-label` prop. Maybe it should be `ariaLabel`?

```
FAIL src/components/make-button-component.spec.tsx
  ● Test suite failed to run

    src/components/make-button-component.spec.tsx:[20](https://github.com/Typeform/embed/actions/runs/8689377407/job/23826940192?pr=646#step:8:21):55 - error TS2322: Type '{ 'aria-label': string; 'data-custom': string; }' is not assignable to type 'Partial<Omit<HTMLButtonElement, "style"> & AdditionalAttributes>'.
      Object literal may only specify known properties, and ''aria-label'' does not exist in type 'Partial<Omit<HTMLButtonElement, "style"> & AdditionalAttributes>'.

    20         <Button id="form-id" foo="bar" buttonProps={{ 'aria-label': 'aria-value-btn', 'data-custom': 'value-custom' }}>
                                                             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

      src/components/make-button-component.tsx:9:3
        9   buttonProps?: ButtonProps
            ~~~~~~~~~~~
        The expected type comes from property 'buttonProps' which is declared here on type 'IntrinsicAttributes & Props & ButtonComponentBaseProps & RefAttributes<unknown>'
```

File: packages/embed-react/src/components/make-button-component.tsx [hunk 2]
```diff
@@ -6,21 +6,22 @@ import React, {
...
 } from 'react'
+import { ButtonProps } from '@typeform/embed'
 
 import { InlineStyle } from './inline-style'
 
 type ButtonComponentBaseProps = {
   id: string
   as?: keyof ReactHTML
-  buttonProps?: HTMLAttributes<HTMLElement> & Record<string, string>
+  buttonProps?: ButtonProps & AriaAttributes
```
Comment (leandro-serafim) [inline packages/embed-react/src/components/make-button-component.tsx:18]: This fixes the failing tests that were missing proper aria typing. 
cc: @mathio

**Comment by typeform-ops-gha** [comment]
*[BOT]* Preview available with hash `85ebfdfd8718a543ed51c94b14a33fe8e7d0914e` [here](https://embed.typeform.com/85ebfdfd8718a543ed51c94b14a33fe8e7d0914e/embed.js).

**Review by mathio — APPROVED** [review]

**Pull request merged**